### PR TITLE
Changed IMemoryList based on Pipelines API Review

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
- Renamed Rest to Next
- Removed interfaces it extended
- Removed CopyTo
- Updated pipelines to use this (shared) interface

@pakrym, @ahsonkhan, @joshfree

### CODE REVIEW

**Comment by davidfowl** [comment]
Getting closer

File: src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs [hunk 1]
```diff
@@ -9,6 +9,118 @@
 
 namespace System.Buffers
 {
+
+    public static class MemoryListExtensions
```
Comment (davidfowl) [inline src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs]: Are we going to use the ReadableBuffer implementation?

Comment (KrzysztofCwalina) [reply src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs]: For? 

BTW, this project is not even referenced by the pipelines. I needed to add all these extension methods below to be able to compile existing code (in this project) after removing CopyTo, and ISequence implementations from IMemoryList. i.e. there is no plan to ship these members.

Comment (davidfowl) [reply src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs]: Ah I see

**Pull request merged**